### ISSUE TITLE
Every client request should be successfully processed

### ISSUE THREAD

**Issue by VladimirRychlitskiy** [DmitriyShchur/esTest#2]
Every client request should be successfully processed even if the user clicks the button during another request execution

**Comment by DmitriyShchur** [comment]
I thought that a main point of this application is an implementation of preventing inconsistent requests.

This issue contradicts the initial requirements and breaks the main point.

Initial Functional requirements were sort of:
_Write an android application with a server emulation.
The application should run 20 requests in the background and receive answers to them.
Create a button "add request out of sequence". The response to the request is an exception. (I decided to use response with error message._

If we implement this issue requirement, what should be a logic for processing a request by button?

**Comment by VladimirRychlitskiy** [comment]
Right, the behavior of the server is totally correct. The error should be processed on the client, and request should be added to the common queue and executed asap instead of just returning an error taking into account the server behavior.

**Comment by DmitriyShchur** [comment]
Implemented in commit https://github.com/DmitriyShchur/esTest/commit/aced8448bf5854a03454155cff6f302886290bfb
